pkgs/applications/networking/instant-messengers/discord/default.nix

```diff
@@ -1,69 +1,115 @@
 {
-  branch ? "stable",
   callPackage,
   fetchurl,
   lib,
   stdenv,
 }:
 let
-  versions =
-    if stdenv.hostPlatform.isLinux then
-      {
-        stable = "0.0.111";
-        ptb = "0.0.161";
-        canary = "0.0.761";
-        development = "0.0.89";
-      }
-    else
-      {
-        stable = "0.0.362";
-        ptb = "0.0.192";
-        canary = "0.0.867";
-        development = "0.0.100";
-      };
-  version = versions.${branch};
-  srcs = rec {
+  variants = rec {
     x86_64-linux = {
-      stable = fetchurl {
-        url = "https://stable.dl2.discordapp.net/apps/linux/${version}/discord-${version}.tar.gz";
-        hash = "sha256-o4U6i223Agtbt1N9v0GO/Ivx68OQcX/N3mHXUX2gruA=";
+      discord = rec {
```

Comment:
Most other packages would put these in a json file or something so that a script can update it